dev/gallery.tsx

```diff
@@ -113,16 +159,49 @@ export function Gallery() {
                     style={styles.cards}
                     className={isMobile ? "perseus-mobile" : ""}
                 >
-                    {questions.map((question, i) => (
-                        <QuestionRenderer
-                            key={i}
-                            question={question}
-                            apiOptions={{
-                                isMobile,
-                                flags: {mafs: mafsFlagsObject},
-                            }}
-                        />
-                    ))}
+                    {questions
+                        .filter((question) => {
+                            if (!search) {
+                                return true;
+                            }
+                            const widgetKey = Object.keys(question.widgets)[0];
+                            const widget = question.widgets[widgetKey];
+                            switch (widget.type) {
```

Comment:
**idea**: Wonder if it'd be worth extracting this matcher function to it's own top-level function to shorten up this component a bit. Right now all the extra indentation is starting to make it difficult to read.